samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -0,0 +1,195 @@
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    /// ## startingNumberConstraints
+    /// The constraints on the empty squares due to numbers already in the 
+    /// puzzle when we start.
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    ///
+    ///
+    /// # Output
+    /// A Tuple with Result and the array of numbers for each empty square
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    ///
+    /// # Remarks
+    /// The inputs and outputs for the following 4x4 puzzle are
+    ///    -----------------
+    ///    |   | 1 | 2 | 3 |         <--- empty square #0
+    ///    -----------------
+    ///    | 2 |   | 0 | 1 |         <--- empty square #1
+    ///    -----------------
+    ///    | 1 | 2 | 3 | 0 |
+    ///    -----------------
+    ///    | 3 |   | 1 | 2 |         <--- empty square #2
+    ///    -----------------
+    ///    emptySquareEdges = [(1, 0),(2, 1)]    
+    ///         empty square #0 can not have the same color/number as empty call #1 
+    ///         empty square #1 and #2 can not have the same color/number (same column)
+    ///    startingNumberConstraints = [(0, 2),(0, 1),(0, 3),(1, 1),(1, 2),(1, 0),(2, 1),(2, 2),(2, 3)]
+    ///         empty square #0 can not have values 2,1,3 because same row/column/2x2grid
+    ///         empty square #1 can not have values 1,2,0 because same row/column/2x2grid
+    ///    Results = [0,3,0] i.e. Empty Square #0 = 0, Empty Square #1 = 3, Empty Square #2 = 0
+    operation SolvePuzzle(numVertices : Int, size : Int, emptySquareEdges : (Int, Int)[], 
+        startingNumberConstraints: (Int, Int)[]) : (Bool, Int[]) {
+        mutable bitsPerColor = 2; // if size == 4x4 grid
+        if (size == 9)
+        {
+            set bitsPerColor = 4; // if size == 9x9 grid
+        }
+        let numIterations = NIterations(bitsPerColor * numVertices);
+        Message($"Running Quantum test with #Vertex = {numVertices}");
+        Message($"   Bits Per Color = {bitsPerColor}");
+        Message($"   emptySquareEdges = {emptySquareEdges}");
+        Message($"   startingNumberConstraints = {startingNumberConstraints}");
+        Message($"   estimated #iterations needed = {numIterations}");
+        Message($"   size of Sudoku grid = {size}x{size}");
+        mutable coloring = new Int[0];
+        if (size == 4) {
+            set coloring = FindColorsWithGrover(numVertices, 2, numIterations, 
+                VertexColoringOracle(numVertices, 2, emptySquareEdges, startingNumberConstraints, _, _));
+        }
+        elif (size == 9) {
+            set coloring = FindColorsWithGrover(numVertices, 4, numIterations, 
+                VertexColoringOracle4Bit9Color(numVertices, emptySquareEdges, startingNumberConstraints, _, _));
+        }
```

Comment:
```suggestion
        let coloring = FindColorsWithGrover(numVertices, bitsPerColor, numIterations, oracle);
```

Reply:
I suggest to move all the if/elif logic up to the same place and use immutable vars as much as possible.